PLFM-3128

kimyen opened the pull request.

kimyen committed:
add new object type ACCESS_CONTROL_LIST
add new queue to hold change messages of ACCESS_CONTROL_LIST type

kimyen committed:
generate change message for creating an acl

kimyen committed:
generate new change messages for update and delete acl

kimyen committed:
Merge remote-tracking branch 'upstream/develop' into kimyen-dev

kimyen committed:
add AclRecordDao and its implementation
add AclWorker to handle ACL change messages

kimyen committed:
add test for AclRecordDAO

kimyen committed:
add test for AclSnapshotWorker

kimyen committed:
change write logic: for each acl record, write it to a temp file and
push the file of 1 record to S3

kimyen committed:
Merge remote-tracking branch 'upstream/develop' into kimyen-dev

kimyen committed:
prevent future bugs on getting all Object Summary from an S3 bucket

kimyen committed:
remove unused code

kimyen committed:
Add an util class to help capture the X-Forwarded-For field
Update the AccessControlListDao to allow getting the acl's dbo using the
aclId (besides using the ownerId and ObjectType)

kimyen committed:
add the acl record and acl resource access record buckets to
StackConstants

kimyen committed:
expose the acl record bucket name and the resource access record bucket
name

kimyen committed:
define the acl record and the resource access record

kimyen committed:
fig bugs on AclRecord and ResourceAccessRecord
get the aclBucketName from StackConfiguration
define a bean for ResourceRecordDao

kimyen committed:
add ResourceAccessRecordDAO and its implementation

kimyen committed:
update AclRecord.json and ResourceAccessRecord.json

kimyen committed:
add test for ResourceAccessRecordDAOImpl

kimyen committed:
refactor: move write methods into a SimpleRecordWriter class that write
any record into a file and push the file to S3

kimyen committed:
update the acl change message to carry the dboId instead of the ownerId

kimyen committed:
update the SimpleRecordWriter to write multiple records into one file

kimyen committed:
update ResourceAccessRecord.json
implement AclSnapshotWorker

kimyen committed:
add test for AclSnapshotWorker

kimyen committed:
update test to wait for S3

kimyen committed:
update get methods

kimyen committed:
update the test

kimyen committed:
Merge remote-tracking branch 'upstream/develop' into kimyen-dev

kimyen committed:
add createdBy, modifiedOn, modifiedBy to acl

kimyen committed:
Revert "add createdBy, modifiedOn, modifiedBy to acl"

This reverts commit 6ddcb7e40ff916198755747ddad43ae9188661dd.

kimyen committed:
remove unsed variable

kimyen committed:
remove fields that always null

kimyen committed:
remove some empty lines

john-hill commented on lib/lib-audit/src/main/java/org/sagebionetworks/audit/utils/SimpleRecordWriter.java:
```diff
@@ -0,0 +1,68 @@
+	
+	public String write(List<T> records) throws IOException {
+		File file = createNewFile();
+		
```
try{

}finally{
file.delete();
}

john-hill commented on lib/lib-audit/src/main/java/org/sagebionetworks/audit/utils/SimpleRecordWriter.java:
```diff
@@ -0,0 +1,68 @@
+	}
+
+	private File createNewFile() throws IOException {
+		File file = new File("temp.csv");
```
use File.createTempFile("temp", ".csv");

john-hill commented on lib/lib-audit/src/test/java/org/sagebionetworks/audit/AclRecordDAOImplTest.java:
```diff
@@ -0,0 +1,64 @@
+	@Autowired
+	private AmazonS3Client s3Client;
+	
+	private String BUCKET_NAME = "dev.acl.record.sagebase.org";
```
will use stack config

kimyen committed:
add test
add try - finally block to delete file and close csvWriter

kimyen committed:
Update SimpleRecordWriter to write a batch of records into a gzip file
Refactor AccessRecordDAOImpl to use SimpleRecordWriter

kimyen committed:
separate the tests for ChangeMessage from the
DBOAccessControlListDaoImplTest

kimyen committed:
add test

kimyen committed:
add test for getOwnerType wth bad AclId

kimyen committed:
refactor all methods in AccessRecordDao into SimpleRecordWorker to be
reused by other record DAO

kimyen committed:
refactor SimpleRecordWorker

kimyen committed:
add method to clean up after each test

kimyen committed:
Merge remote-tracking branch 'upstream/develop' into kimyen-dev

kimyen committed:
Update ObjectCSVReader to read enum types

kimyen committed:
update and add more tests for AclSnapshotWorkerIntegrationTest

kimyen committed:
add unit test for AclSnapshotWorker

kimyen committed:
add case where ChangeMessage is null

kimyen committed:
add comments

kimyen committed:
test protected methods

kimyen committed:
use another method to compare acl records

kimyen committed:
handle reading enum

kimyen committed:
update xml file

kimyen committed:
add an empty constructor to help creating bean

kimyen committed:
add getAclId method using ownerId and ownerType
use queryForLong and queryForObject to get one field instead of getting
the complete Object
update test

kimyen committed:
rename SimpleRecordWorker to ObjectCSVDAO

kimyen committed:
add listAllKeys for some DAOs (used in every test)

kimyen committed:
refactor code in AclSnapshotWorker

kimyen committed:
use timestamp as a param in buildMessage

kimyen committed:
organize import

kimyen committed:
normalize the ResourceAccessRecord

kimyen committed:
add more test

kimyen committed:
remove handling reading Set Type

kimyen committed:
separate before and after to make sure that the test clean up after
itself

kimyen committed:
refactor code

kimyen committed:
refactor code

kimyen committed:
refactor code

kimyen committed:
remove unused code

kimyen committed:
Merge remote-tracking branch 'upstream/develop' into kimyen-dev

kimyen committed:
add deleteAllAcl method to clean up the database after tests
refactor DBOAccessControlListDAOChangeMessageTest

kimyen committed:
rewrite the logic of the worker to handle DELETE differently

kimyen committed:
use test utils to refactor code

kimyen committed:
refactor integration test

kimyen committed:
refactor

kimyen committed:
refactor

kimyen committed:
do not set timestamp in buildMessage method that does not take timestamp
as a parameter

kimyen committed:
always write an aclRecord if the acl is not null

kimyen committed:
Merge remote-tracking branch 'upstream/develop' into kimyen-dev

kimyen committed:
refactor util method

kimyen committed:
check the etag after getting the acl
